src/App.tsx

```diff
@@ -1,93 +1,224 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { getVisibleTodos } from './helpers/getVisibleTodos';
+import {
+  addTodo,
+  getTodos,
+  deleteTodo,
+  updateTodo,
+} from './api/todos';
+import { USER_ID, getUser } from './api/user';
+import { User } from './types/User';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [user, setUser] = useState<User | null>(null);
+  const [title, setTitle] = useState('');
+  const [filterBy, setFilterBy] = useState<Status>(Status.ALL);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const clearError = () => {
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  useEffect(() => {
+    const loadTodos1 = localStorage.getItem('todos');
+
+    if (loadTodos1) {
+      setTodos(JSON.parse(loadTodos1));
+    }
+  }, []);
+
+  useEffect(() => {
+    localStorage.setItem('todos', JSON.stringify(todos));
+  }, [todos]);
+
+  const getUserName = async () => {
+    const loadedUser = await getUser(USER_ID);
+
+    setUser(loadedUser);
+  };
+
+  const loadTodos = async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch (error) {
+      setErrorMessage((error as Error).message);
+
+      clearError();
+    }
+  };
+
+  useEffect(() => {
+    getUserName();
+    loadTodos();
+  }, []);
+
+  const visibleTodos = getVisibleTodos(todos, filterBy);
+
+  const handleAddNewTodo = async () => {
+    try {
+      if (title.trim()) {
+        const newTodo = {
+          // 1000 division is needed because server returns 500 otherwise.
+          id: Math.trunc(+new Date() / 1000),
```

Comment:
Generating id based on time may lead to collisions, prefer using a [randomUUID](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID) to generate unique identifiers.

Reply:
Not resolved

Reply:
![Screenshot 2023-06-18 at 20 57 09](https://github.com/mate-academy/react_todo-app/assets/81584108/16af0147-8983-4b03-ae7a-f63982d700bd)
I used new Date() because it was in the task description.
I can't use randomUUID method since the ids are stored as numbers on the server. So, I create a new todo with id: 0. Then the API generates a unique id on its side.